### PR TITLE
Add Scheduled Updates /active endpoint

### PR DESCRIPTION
**Pull request by zaerl**
Context: p1714398995145929-slack-C01A60HCGUA

## Proposed changes:

* Removed the `active` field from the POST endpoints.
* Added new `/(?P<schedule_id>[\w]+)/active` endpoint.
* Adjusted unit tests according
* Fix: wrong check on `test_run_inactive_schedule`

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:
* Please apply the PR using the `jetpack-downloader`
* Install D147343-code
* Follow instructions found here https://github.com/Automattic/jetpack/pull/37110
* In the https://developer.wordpress.com/docs/api/console/ make two calls to `POST /sites/__SITE__/update-schedules/__ID__/active?active=1` and `active=0`. Check that everything work as expected (the scheduled update should not run)

### CODE REVIEW

**Comment by obenland** [comment]
I'll try to get #37008 in today. With that in place, let's break out this new endpoint into a separate file as well

**Review by ouikhuan — COMMENTED** [review]
We'll also need to trigger `update_option_cron` like what we did here: https://github.com/Automattic/jetpack/blob/trunk/projects/packages/scheduled-updates/src/class-scheduled-updates.php#L79-L84 

So when the active status is updated, it'll also reflected in `jetpack_scheduled_plugins_update` option. 

On WPCOM side, we'll need to add this into jetpack-bridge.php so it can be called via public-api and use in Calypso later. What do you think?

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	 */
+	public function update_item( $request ) {
+		$events = wp_get_scheduled_events( Scheduled_Updates::PLUGIN_CRON_HOOK );
+
+		if ( empty( $events[ $request['schedule_id'] ] ) ) {
+			return new WP_Error( 'rest_invalid_schedule', __( 'The schedule could not be found.', 'jetpack-scheduled-updates' ), array( 'status' => 404 ) );
+		}
+
+		Scheduled_Updates_Active::update( $request['schedule_id'], (bool) $request['active'] );
+		Scheduled_Updates::update_option_cron();
```
Comment (ouikhuan) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php]: Out of curiosity, is there any reason we decide not to handle this in a hook like we do for others?

Comment (zaerl) [reply projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php]: Not in particular 🙂 , I moved it to cron callbacks https://github.com/Automattic/jetpack/pull/37130/commits/fda1965412cc11d6b11515fa3754891401fd9fc0

**Review by ouikhuan — APPROVED** [review]
Works as expected, nice one!

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	 *
+	 * @var string
+	 */
+	public $rest_base = 'update-schedules';
+
+	/**
+	 * WPCOM_REST_API_V2_Endpoint_Update_Schedules_Active constructor.
+	 */
+	public function __construct() {
+		// Priority 9 to run before the main endpoint and avoid "/active" being treated as a schedule ID.
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php:35]: Would it be treated as a schedule ID? Looking at the route, there should be one provided

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => array(
+						'active' => array(
+							'description'       => 'Whether the schedule is active.',
+							'type'              => 'boolean',
+							'required'          => true,
+							'sanitize_callback' => 'rest_sanitize_boolean',
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php:56]: I would have expected defining the type as boolean to trigger `rest_sanitize_boolean` internally. Is that not the case?

File: projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	 * @return WP_REST_Response|WP_Error The updated active value or a WP_Error if the schedule could not be found.
+	 */
+	public function update_item( $request ) {
+		$events = wp_get_scheduled_events( Scheduled_Updates::PLUGIN_CRON_HOOK );
+
+		if ( empty( $events[ $request['schedule_id'] ] ) ) {
+			return new WP_Error( 'rest_invalid_schedule', __( 'The schedule could not be found.', 'jetpack-scheduled-updates' ), array( 'status' => 404 ) );
+		}
+
+		Scheduled_Updates_Active::update( $request['schedule_id'], (bool) $request['active'] );
```
Comment (obenland) [inline projects/packages/scheduled-updates/src/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-update-schedules-active.php:91]: What does `$request['active']` come out as, that it has to be cast to boolean?

**Pull request merged**